README.md

```diff
@@ -184,9 +213,49 @@ python3 script.py apply example-data.json
 }
 ```
 
+## MQ clients
+
+The MQ clients implement sender & receiver functionality, to provide the configured node during performance and throughput testing.
+For each MQ, a docker image with the clients for that MQ is built. When starting the container, an HTTP server is started. 
+The server exposes the following endpoints:
+
+* `GET /metrics`
+* `GET /in-progress` - a JSON array of names of processes (either `sender` or `reciever`) that have been started by this server
+* `POST /init` - initialize thq MQ (e.g. creating topics), as described by the given configuration; ideally should be run once before starting a sender/receiver
+* `POST /start/receiver` and `POST /start/sender` - start a process of receiving/sending, as described by the given configuration
+
+The server by default binds to `0.0.0.0:8080`, but this can be customised by setting the `http.host` and `http.port` environment variables.
+
+The structure of the config JSON is:
+
+```json
+{
+  "testId": "test",
+  "testLengthSeconds": 10,
+  "msgsPerSecond": 10,
+  "msgSizeBytes": 100,
+  "batchSizeSend": 1,
+  "batchSizeReceive": 1,
+  "maxSendInFlight": 1,
+  "mqConfig":{  
+    
+  }
+}
+```
+
+See docs in `Config` for an explanation of the parameters.
+
+### Kafka
+
+To test the senders/receivers using Kafka, go to `docker/kafka` and run `docker-compose up`. This will start zookeeper, the kafka broker,
+Confluent's control center (available at `http://localhost:9021`), and the mqperf server.
```

Comment:
I'm not really familiar with scala or it's build plugins so I think adding a command how to publish a new docker image using scala build plugin would be nice. Unless there will be no need to publish new images for users of this project.

Reply:
Good point, added